cmd/process-agent/app/status.go

```diff
@@ -0,0 +1,289 @@
+	if err != nil {
+		return coreStatus{}, fmt.Errorf("config error: %s", err.Error())
+	}
+
+	statusEndpoint := fmt.Sprintf("http://%s/agent/status", addressPort)
+	b, err := util.DoGet(httpClient, statusEndpoint)
+	if err != nil {
+		return s, connectionError{err}
+	}
+
+	err = json.Unmarshal(b, &s)
+	return
+}
+
+func getExpvars() (s processExpvars, err error) {
+	ipcAddr, err := ddconfig.GetIPCAddress()
+	if err != nil {
+		return processExpvars{}, fmt.Errorf("config error: %s", err.Error())
+	}
+
+	port := ddconfig.Datadog.GetInt("process_config.expvar_port")
+	if port <= 0 {
+		_ = log.Warnf("Invalid process_config.expvar_port -- %d, using default port %d\n", port, ddconfig.DefaultProcessExpVarPort)
+		port = ddconfig.DefaultProcessExpVarPort
+	}
+	expvarEndpoint := fmt.Sprintf("http://%s:%d/debug/vars", ipcAddr, port)
+	b, err := util.DoGet(httpClient, expvarEndpoint)
+	if err != nil {
+		return s, connectionError{err}
+	}
+
+	err = json.Unmarshal(b, &s)
+	return
+}
+
+func getStatus() (status, error) {
+	coreStatus, err := getCoreStatus()
+	if err != nil {
+		return status{}, err
+	}
+
+	processStatus, err := getExpvars()
+	if err != nil {
+		return status{}, err
+	}
+
+	s := status{
+		Date:    float64(time.Now().UnixNano()),
+		Core:    coreStatus,
+		Expvars: processStatus,
+	}
+	return s, nil
+}
+
+func writeNotRunning(w io.Writer) {
+	_, err := fmt.Fprint(w, notRunning)
+	if err != nil {
+		_ = log.Error(err)
+	}
+}
+
+func writeError(w io.Writer, e error) {
+	tpl, err := template.New("").Funcs(ddstatus.Textfmap()).Parse(errorMessage)
+	if err != nil {
+		_ = log.Error(err)
+	}
+
+	err = tpl.Execute(w, e)
+	if err != nil {
+		_ = log.Error(err)
+	}
+}
+
+// getAndWriteStatus calls the status server and writes it to `w`
+func getAndWriteStatus(w io.Writer, options ...statusOption) {
+	status, err := getStatus()
+	if err != nil {
+		switch err.(type) {
+		case connectionError:
+			writeNotRunning(w)
+		default:
+			writeError(w, err)
+		}
+		return
+	}
+	for _, option := range options {
+		option(&status)
+	}
+
+	tpl, err := template.New("").Funcs(ddstatus.Textfmap()).Parse(statusTemplate)
+	if err != nil {
+		_ = log.Error(err)
+	}
+
+	err = tpl.Execute(w, status)
+	if err != nil {
+		_ = log.Error(err)
+	}
+}
+
+// StatusCmd returns a cobra command that prints the current status
+func StatusCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "status",
+		Short: "Print the current status",
+		Long:  ``,
+		RunE:  runStatus,
+	}
+}
+
+func runStatus(cmd *cobra.Command, _ []string) error {
+	err := config.LoadConfigIfExists(cmd.Flag("config").Value.String())
+	if err != nil {
+		writeError(os.Stdout, err)
+	}
+
+	err = ddconfig.SetupLogger(
+		"process",
+		ddconfig.Datadog.GetString("log_level"),
+		ddconfig.Datadog.GetString("process_config.log_file"),
+		ddconfig.GetSyslogURI(),
+		ddconfig.Datadog.GetBool("syslog_rfc"),
+		ddconfig.Datadog.GetBool("log_to_console"),
+		ddconfig.Datadog.GetBool("log_format_json"),
+	)
```

Comment:
Since we're going to run the `status` command as a separate process, I wonder if we should use settings similar to core agent commands:
https://github.com/DataDog/datadog-agent/blob/756f0ac9227a674cdec9595ca39e30ba51a118d3/cmd/agent/app/status.go#L60

We may not want to pollute the `process-agent.log` file with it.

Reply:
I agree, I think that would be smart